modules/cache-common/src/main/java/org/opensearch/common/cache/stats/TieredSpilloverCacheStatsHolder.java

```diff
@@ -164,7 +166,31 @@ public void decrementItems(List<String> dimensionValues) {
         super.decrementItems(dimensionValues);
     }
 
-    void setDiskCacheEnabled(boolean diskCacheEnabled) {
+    public void setDiskCacheEnabled(boolean diskCacheEnabled) {
         this.diskCacheEnabled = diskCacheEnabled;
     }
+
+    @Override
+    public void removeDimensions(List<String> dimensionValues) {
```

Comment:
Can't we just reuse removeDimensions method from DefaultCacheStatsHolder? Both look exactly same.

Reply:
For the TSC we check for `assert dimensionValues.size() == dimensionNames.size() - 1` rather than `dimensionNames.size()`, so that the base case is the node whose children are the individual tier nodes, but besides the check they're otherwise the same.